lib/providers/theme/themeProvider.tsx

```diff
@@ -24,6 +24,21 @@ export const ThemeProvider = ({ children, theme }: ThemeProviderProps) => {
         const theme = selectedTheme || (systemPrefersDark ? "dark" : "light");
 
         setCurrentTheme(theme);
+
+        const link = document.createElement("link");
+        link.rel = "stylesheet";
+
+        if (theme === "light") {
+            link.href = "/lib/styles/quill/light.scss";
+        } else {
+            link.href = "/lib/styles/quill/dark.scss";
+        }
+
+        document.head.appendChild(link);
+
+        return () => {
+            document.head.removeChild(link);
+        };
```

Comment:
I noticed that this theme provider doesn't detect themes correctly, and with these changes, it's always dark for me.
Even though '**light theme**' is selected in storybook, it's still **dark** perhaps because my MacOS system prefers dark:
<img width="1192" alt="Screenshot 2024-04-24 at 5 27 02 PM" src="https://github.com/deriv-com/quill-ui/assets/103177211/a52cae7e-b1c8-48ff-851d-68407647059d">

Also, I would recommend avoiding working with DOM directly as React suggests, and not append stylesheets to it but instead use an approach that we have in deriv-app:
importing the stylesheet into an entry file of the application and switching between themes based on the classNames **'theme--light' & 'theme--dark'**.
Deriv-app example:
<img width="851" alt="Screenshot 2024-04-24 at 5 35 09 PM" src="https://github.com/deriv-com/quill-ui/assets/103177211/c3f98768-6a90-442b-bad3-402cafc23fde">

We can still have separate files for light and dark styles if you need but they would be nested under those classnames:
```
:root {
    .theme--light {
```
@prince-deriv  Is it possible?

Reply:
@maryia-deriv  thanks for the check and approach, suggestion. I have tried a closer approach and it did work

Reply:
@prince-deriv 🥲 unfortunately, the issue with dark theme always enforced still persists for me.
I guess my system prefers dark, so ThemeProvider takes it and it doesn't care which theme is selected in the app.
<img width="1149" alt="Screenshot 2024-04-25 at 1 52 41 PM" src="https://github.com/deriv-com/quill-ui/assets/103177211/c46cfd1b-69a0-412d-b412-975021a916cd">

Reply:
@maryia-deriv  for storybook it will be a different fix. We will have a PR for that. Can you try running the app `npm run dev` and see if the theme works for you properly?